### DISCUSSION TITLE
Need help with creating a scriptlet, round three

### DISCUSSION THREAD

**Discussion by partingscientist** [uBlockOrigin/uBlock-issues#2785] [Q&A] [answered]
I tried converting an existing scriptlet of mine into the new named-function form.

```javascript
/// twitch-claim-bonus.js
/// world isolated
function twitchClaimBonus() {
    "use strict";
    console.log("Checking for button container...");
    (() => {
        const callback = (_, observer) => {
            const elements = document.getElementsByClassName("chat-input__buttons-container");
            if (elements.length > 0) {
                console.log("Button container found! Initiating auto-claim...");
                observer.disconnect();
                checkButton(elements[0]);
            }
        };
        const observer = new MutationObserver(callback);
        observer.observe(document.documentElement, {subtree: true, childList: true});
    })();
    function checkButton(element) {
        const callback = (_, observer) => {
            try {
                document.getElementsByClassName("claimable-bonus__icon")[0].click();
                console.log(`Bonus point claim succeed! - ${Date()}`);
            } catch (error) {
                console.log(`Bonus point button isn't found! - ${Date()}`);
            }
            observer.takeRecords();
        };
        const observer = new MutationObserver(callback);
        observer.observe(element, {subtree: true, childList: true});
    }
}
```

The scriptlet does not seem to be injected when I open Twitch (i.e. I cannot find its source using the debugger). Any possible reason(s) as to why? Thank you in advance.

(Firefox 116.0.3, uBlock Origin 1.51.1rc1)

**Comment by gorhill** [comment]
What the URL for that scriptlet? What's the filter you are using?

**Reply by partingscientist** [reply]
Here you go: [snipped]
For the example above, [snipped]

**Reply by gorhill** [reply, marked as answer]
The scriptlet you list here is different than the one in your file -- please always provide the real actual data which is at the heart of the issue you are reporting -- I would have wasted time investigating to no avail if I didn't ask you to provide the actual, real repro steps.

You are inserting comment `/* ... */` in between the scriptlet descriptors and the scriptlet function, this is your issue.

uBO does not parse JS, and as a result it has strict expectations regarding how scriptlets are defined. Anything which follows the scriptlet descriptors (`///`) which does not start with `//` will be deemed part of the scriptlet body. As a result, your scriptlet body starts with a comment, which prevent uBO from detecting that your scriptlet itself starts with `function(...) { ...`, which prevents uBO from properly patching your scriptlet for execution. Don't use anything between scriptlet descriptors and `function(...) { ...`, except for lines starting with `// ...`, which will be thrown away by uBO.

Also be aware that I make no guarantee regarding dependencies you are using in your scriptlets, I may change these at any time, they are primarily to be used for internal purpose, I just made them available for the adventurous.

**Reply by partingscientist** [reply]
Yeah that is my bad. I didn't expect things to be different (should have considered it) with the named-function form, considering adding JSDoc between descriptor and the scriptlet body worked with the text-based one. My apologies.

**Comment by uBlock-user** [comment]
Why two mutation observers ?

**Reply by partingscientist** [reply]
That was the original solution that I come up with for https://github.com/uBlockOrigin/uBlock-issues/discussions/2514#discussioncomment-5175652. Feel free if you have any suggestion for it!
